### DISCUSSION TITLE
Get return function task

### DISCUSSION THREAD

**Discussion by jonatasoli** [ag2ai/faststream#1342] [Q&A] [answered]
Hi,

I have asked from a problem with the return function.
I have a function below in my services.py

```python
    cart_uuid: str,
    payment_method: str,
    payment_gateway: str,
    user: Any,
    bootstrap: Any = Depends(get_bootstrap),
    checkout_task = await bootstrap.message.broker.publish(
        {
            'cart_uuid': uuid,
            'payment_gateway': cache_cart.gateway_provider,
            'payment_method': cache_cart.payment_method,
            'user': user,
        },
        queue=RabbitQueue('checkout'),
    )
```

checkout_task is a variable receive the return this checkout task ex:.
task.py

@worker.subscriber
async def checkout(
    cart_uuid: str,
    payment_method: str,
    payment_gateway: str,
    user: Any,
    bootstrap: Any = Depends(get_bootstrap),
) -> dict:

    return { 'status': ok, 'processing''ok' }
```

I need get the return in checkout task but this form I don´t receive this return, I go from propan lib in this case I use `callback` flag to inform this task is need wait the return, but I don't see how done this form in faststream.

**Comment by Lancetnik** [comment, marked as answer]
Are you looking for "rpc"? https://faststream.airt.ai/latest/rabbit/rpc/

**Reply by jonatasoli** [reply]
No, in my case I'd like only wait task finish and respond for function with publish.

**Reply by jonatasoli** [reply]
I try rpc and found for me thanks for the help =D
